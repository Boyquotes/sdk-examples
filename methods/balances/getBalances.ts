import { ExecuteMethod } from "../../executeMethod";

export const executeMethod: ExecuteMethod = async ({ cedeSDK, exchangeInstanceId }) => {
  const { getBalances } = cedeSDK.api;

  const start = new Date();
  const balances = await getBalances({ exchangeInstanceId });
  const end = new Date();
  console.log(`Execution time: ${end.getTime() - start.getTime()}ms`);

  return balances;
};
