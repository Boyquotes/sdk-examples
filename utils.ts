import CedeSDK from "@cedelabs/demo-sdk";
import { exchanges } from "./exchanges";

const SDK_MODE = "MOCK";
const SELECTED_EXCHANGE = "bybit";

const clientId = "sdk-***"; // update with your clientId if using with "REAL" SDK_MODE

export const cedeSDK = new CedeSDK(SDK_MODE ?? "MOCK");

export const initExchange = async (specificExchange = "coinbase") => {
  const selectedExchange = SELECTED_EXCHANGE ?? specificExchange;

  function assertIsExchangeKey(key: string): key is keyof typeof exchanges {
    return key in exchanges;
  }

  if (!assertIsExchangeKey(selectedExchange)) {
    throw new Error(`Exchange ${selectedExchange} not found`);
  }

  const exchange = exchanges[selectedExchange];
  const registeredExchange = await cedeSDK.api.registerExchangeInstance({
    exchangeId: SELECTED_EXCHANGE ?? specificExchange,
    ...exchange,
  });

  return registeredExchange.exchangeInstanceId;
};
