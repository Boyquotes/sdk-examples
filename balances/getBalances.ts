import { cedeSDK, initExchange } from "../utils";

export const executeMethod = async () => {
  const exchangeInstanceId = await initExchange();
  const { api } = cedeSDK;
  const { getBalances } = api;

  const balances = await getBalances({ exchangeInstanceId });

  return balances;
};
