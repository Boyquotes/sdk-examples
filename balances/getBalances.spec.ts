import { initExchange } from "../utils";
import { executeMethod } from "./getBalances";

describe("getBalances", () => {
  beforeAll(async () => {
    await initExchange();
  });

  it("it should return balances", async () => {
    const result = await executeMethod();

    expect(result).toMatchSnapshot({
      data: {
        DOT: {
          freeBalance: expect.any(Number),
          refFreeBalance: expect.any(Number),
          refTotalBalance: expect.any(Number),
          refUsedBalance: 0,
          spot: {
            freeBalance: expect.any(Number),
            refFreeBalance: expect.any(Number),
            refTotalBalance: expect.any(Number),
            refUsedBalance: 0,
            totalBalance: expect.any(Number),
            usedBalance: 0,
          },
          totalBalance: expect.any(Number),
          usedBalance: 0,
        },
        ETH: {
          freeBalance: expect.any(Number),
          refFreeBalance: expect.any(Number),
          refTotalBalance: expect.any(Number),
          refUsedBalance: 0,
          spot: {
            freeBalance: expect.any(Number),
            refFreeBalance: expect.any(Number),
            refTotalBalance: expect.any(Number),
            refUsedBalance: 0,
            totalBalance: expect.any(Number),
            usedBalance: 0,
          },
          totalBalance: expect.any(Number),
          usedBalance: 0,
        },
        LINK: {
          freeBalance: expect.any(Number),
          refFreeBalance: expect.any(Number),
          refTotalBalance: expect.any(Number),
          refUsedBalance: 0,
          spot: {
            freeBalance: expect.any(Number),
            refFreeBalance: expect.any(Number),
            refTotalBalance: expect.any(Number),
            refUsedBalance: 0,
            totalBalance: expect.any(Number),
            usedBalance: 0,
          },
          totalBalance: expect.any(Number),
          usedBalance: 0,
        },
        USDT: {
          freeBalance: expect.any(Number),
          refFreeBalance: expect.any(Number),
          refTotalBalance: expect.any(Number),
          refUsedBalance: 0,
          spot: {
            freeBalance: NaN,
            refFreeBalance: NaN,
            refTotalBalance: NaN,
            refUsedBalance: 0,
            totalBalance: NaN,
            usedBalance: 0,
          },
          totalBalance: expect.any(Number),
          usedBalance: 0,
        },
      },
      hydrate: false,
      key: expect.any(String),
    });
  });
});
