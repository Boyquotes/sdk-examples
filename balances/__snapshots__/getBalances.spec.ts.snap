// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getBalances it should return balances 1`] = `
{
  "data": {
    "DOT": {
      "freeBalance": Any<Number>,
      "refFreeBalance": Any<Number>,
      "refTotalBalance": Any<Number>,
      "refUsedBalance": 0,
      "spot": {
        "freeBalance": Any<Number>,
        "refFreeBalance": Any<Number>,
        "refTotalBalance": Any<Number>,
        "refUsedBalance": 0,
        "totalBalance": Any<Number>,
        "usedBalance": 0,
      },
      "totalBalance": Any<Number>,
      "usedBalance": 0,
    },
    "ETH": {
      "freeBalance": Any<Number>,
      "refFreeBalance": Any<Number>,
      "refTotalBalance": Any<Number>,
      "refUsedBalance": 0,
      "spot": {
        "freeBalance": Any<Number>,
        "refFreeBalance": Any<Number>,
        "refTotalBalance": Any<Number>,
        "refUsedBalance": 0,
        "totalBalance": Any<Number>,
        "usedBalance": 0,
      },
      "totalBalance": Any<Number>,
      "usedBalance": 0,
    },
    "LINK": {
      "freeBalance": Any<Number>,
      "refFreeBalance": Any<Number>,
      "refTotalBalance": Any<Number>,
      "refUsedBalance": 0,
      "spot": {
        "freeBalance": Any<Number>,
        "refFreeBalance": Any<Number>,
        "refTotalBalance": Any<Number>,
        "refUsedBalance": 0,
        "totalBalance": Any<Number>,
        "usedBalance": 0,
      },
      "totalBalance": Any<Number>,
      "usedBalance": 0,
    },
    "USDT": {
      "freeBalance": Any<Number>,
      "refFreeBalance": Any<Number>,
      "refTotalBalance": Any<Number>,
      "refUsedBalance": 0,
      "spot": {
        "freeBalance": NaN,
        "refFreeBalance": NaN,
        "refTotalBalance": NaN,
        "refUsedBalance": 0,
        "totalBalance": NaN,
        "usedBalance": 0,
      },
      "totalBalance": Any<Number>,
      "usedBalance": 0,
    },
  },
  "hydrate": false,
  "key": Any<String>,
}
`;
